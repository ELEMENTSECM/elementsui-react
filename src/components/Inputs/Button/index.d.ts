import * as React from 'react';

export interface ButtonProps {
	/**
	 * Button label
	 */
	label?: string;
	/**
	 * Button is disabled
	 */
	disabled?: boolean;
	/**
	 * Mouse click event handler
	 */
	onCLick?: (...args: any[]) => any;
	/**
	 * Primary button
	 */
	isPrimary?: boolean;
}

export const Button: React.SFC<ButtonProps>;
