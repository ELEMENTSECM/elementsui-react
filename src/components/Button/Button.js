import * as React from 'react';
import { DefaultButton } from 'office-ui-fabric-react/lib/Button';
import PropTypes from 'prop-types';

/** Button example */
function Button({ label, disabled = false }) {
	return (
		<DefaultButton disabled={disabled} primary={true}>
			{label}
		</DefaultButton>
	);
}

Button.propTypes = {
	/** Button label */
	label: PropTypes.string,
	/** Button is disabled */
	disabled: PropTypes.bool
};

export default Button;
