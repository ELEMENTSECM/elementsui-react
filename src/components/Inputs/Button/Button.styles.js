export const getStyles = props => {
	const { className, theme } = props;
	const { palette, semanticColors } = theme;

	return {
		root: ['ms-df', {}, className]
	};
};
