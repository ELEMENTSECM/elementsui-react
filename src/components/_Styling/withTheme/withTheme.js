import { Fabric } from 'office-ui-fabric-react/lib/Fabric';
import { loadTheme, AnimationClassNames } from 'office-ui-fabric-react/lib/Styling';
import * as React from 'react';
import PropTypes from 'prop-types';

function withTheme(WrappedComponent, palette) {
	return class extends React.Component {
		constructor(props) {
			super(props);
			loadTheme({
				palette
			});
		}

		render() {
			return (
				<Fabric>
					<WrappedComponent {...this.props} />
				</Fabric>
			);
		}
	};
}

withTheme.propTypes = {
	/**
	 * Color code for themeDarker.
	 */
	themeDarker: PropTypes.string,
	/**
	 * Color code for themeDark.
	 */
	themeDark: PropTypes.string,
	/**
	 * Color code for themeDarkAlt.
	 */
	themeDarkAlt: PropTypes.string,
	/**
	 * Color code for themePrimary.
	 */
	themePrimary: PropTypes.string,
	/**
	 * Color code for themeSecondary.
	 */
	themeSecondary: PropTypes.string,
	/**
	 * Color code for themeTertiary.
	 */
	themeTertiary: PropTypes.string,
	/**
	 * Color code for themeLight.
	 */
	themeLight: PropTypes.string,
	/**
	 * Color code for themeLighter.
	 */
	themeLighter: PropTypes.string,
	/**
	 * Color code for themeLighterAlt.
	 */
	themeLighterAlt: PropTypes.string,
	/**
	 * Color code for the strongest color, which is black in the default theme. This is a very light color in inverted themes.
	 */
	black: PropTypes.string,
	/**
	 * Color code for blackTranslucent40.
	 */
	blackTranslucent40: PropTypes.string,
	/**
	 * Color code for neutralDark.
	 */
	neutralDark: PropTypes.string,
	/**
	 * Color code for neutralPrimary.
	 */
	neutralPrimary: PropTypes.string,
	/**
	 * Color code for neutralPrimaryAlt.
	 */
	neutralPrimaryAlt: PropTypes.string,
	/**
	 * Color code for neutralSecondary.
	 */
	neutralSecondary: PropTypes.string,
	/**
	 * Color code for neutralTertiary.
	 */
	neutralTertiary: PropTypes.string,
	/**
	 * Color code for neutralTertiaryAlt.
	 */
	neutralTertiaryAlt: PropTypes.string,
	/**
	 * Color code for neutralQuaternary.
	 */
	neutralQuaternary: PropTypes.string,
	/**
	 * Color code for neutralQuaternaryAlt.
	 */
	neutralQuaternaryAlt: PropTypes.string,
	/**
	 * Color code for neutralLight.
	 */
	neutralLight: PropTypes.string,
	/**
	 * Color code for neutralLighter.
	 */
	neutralLighter: PropTypes.string,
	/**
	 * Color code for neutralLighterAlt.
	 */
	neutralLighterAlt: PropTypes.string,
	/**
	 * Color code for the accent.
	 */
	accent: PropTypes.string,
	/**
	 * Color code for the softest color, which is white in the default theme. This is a very dark color in dark themes.
	 * This is the page background.
	 */
	white: PropTypes.string,
	/**
	 * Color code for whiteTranslucent40
	 */
	whiteTranslucent40: PropTypes.string,
	/**
	 * Color code for yellow.
	 */
	yellow: PropTypes.string,
	/**
	 * Color code for yellowLight.
	 */
	yellowLight: PropTypes.string,
	/**
	 * Color code for orange.
	 */
	orange: PropTypes.string,
	/**
	 * Color code for orangeLight.
	 */
	orangeLight: PropTypes.string,
	/**
	 * Color code for orangeLighter.
	 */
	orangeLighter: PropTypes.string,
	/**
	 * Color code for redDark.
	 */
	redDark: PropTypes.string,
	/**
	 * Color code for red.
	 */
	red: PropTypes.string,
	/**
	 * Color code for magentaDark.
	 */
	magentaDark: PropTypes.string,
	/**
	 * Color code for magenta.
	 */
	magenta: PropTypes.string,
	/**
	 * Color code for magentaLight.
	 */
	magentaLight: PropTypes.string,
	/**
	 * Color code for purpleDark.
	 */
	purpleDark: PropTypes.string,
	/**
	 * Color code for purple.
	 */
	purple: PropTypes.string,
	/**
	 * Color code for purpleLight.
	 */
	purpleLight: PropTypes.string,
	/**
	 * Color code for blueDark.
	 */
	blueDark: PropTypes.string,
	/**
	 * Color code for blueMid.
	 */
	blueMid: PropTypes.string,
	/**
	 * Color code for blue.
	 */
	blue: PropTypes.string,
	/**
	 * Color code for blueLight.
	 */
	blueLight: PropTypes.string,
	/**
	 * Color code for tealDark.
	 */
	tealDark: PropTypes.string,
	/**
	 * Color code for teal.
	 */
	teal: PropTypes.string,
	/**
	 * Color code for tealLight.
	 */
	tealLight: PropTypes.string,
	/**
	 * Color code for greenDark.
	 */
	greenDark: PropTypes.string,
	/**
	 * Color code for green.
	 */
	green: PropTypes.string,
	/**
	 * Color code for greenLight.
	 */
	greenLight: PropTypes.string
};

export default withTheme;
