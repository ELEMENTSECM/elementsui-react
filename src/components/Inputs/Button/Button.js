import * as React from 'react';
import PropTypes from 'prop-types';
import { DefaultButton } from 'office-ui-fabric-react/lib/Button';

/** Button example */
function Button({ label, disabled = false, onClick, isPrimary }) {
	return (
		<DefaultButton disabled={disabled} primary={isPrimary} onClick={onClick}>
			{label}
		</DefaultButton>
	);
}

Button.propTypes = {
	/** Button label */
	label: PropTypes.string,
	/** Button is disabled */
	disabled: PropTypes.bool,
	/** Mouse click event handler */
	onCLick: PropTypes.func,
	/** Primary button */
	isPrimary: PropTypes.bool
};

export default Button;
