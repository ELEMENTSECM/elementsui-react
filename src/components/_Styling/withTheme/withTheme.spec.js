import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../Inputs/Button';
import withTheme from './withTheme';

describe('withTheme', () => {
	test('Should render Button with theme applied', () => {
		const StyledButton = withTheme(Button, {
			buttonBackground: 'blue',
			buttonBackgroundHovered: 'red',
			buttonText: 'white',
			buttonTextHovered: 'teal'
		});
		const wrapper = shallow(
			<StyledButton htmlId="defaultBtn" label="Default button" isPrimary={false} />
		);
		expect(wrapper).toMatchSnapshot();
	});
});
