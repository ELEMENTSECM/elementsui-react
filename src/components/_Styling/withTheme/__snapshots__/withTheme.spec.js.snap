// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withTheme Should render Button with theme applied 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_class
    htmlId="defaultBtn"
    isPrimary={false}
    label="Default button"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Button
        htmlId="defaultBtn"
        isPrimary={false}
        label="Default button"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "htmlId": "defaultBtn",
        "isPrimary": false,
        "label": "Default button",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Button
          htmlId="defaultBtn"
          isPrimary={false}
          label="Default button"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "htmlId": "defaultBtn",
          "isPrimary": false,
          "label": "Default button",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
